gene/insert.go

```diff
@@ -0,0 +1,326 @@
+package gene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/numbers"
+	"log"
+)
+
+// Insertion adds bases to the Gene, predicts the effect, and updates the Gene struct to reflect the change.
+// The position should be the base-zero genomic coordinate of the base directly BEFORE the inserted bases.
+func Insertion(g *Gene, genomePos int, alt []dna.Base) (EffectPrediction, error) {
+	var answer EffectPrediction
+	answer.StopDist = -1
+	var genomeIndexPos int
+	var err error
+
+	genomeIndexPos, err = insertionPreRunChecks(g, genomePos, alt)
+
+	// Update genome sequence
+	g.genomeSeq = dna.Insert(g.genomeSeq, genomeIndexPos+1, alt)
+
+	// Update CDS starts and ends
+	for idx, val := range g.cdsStarts {
+		if val <= genomeIndexPos && g.cdsEnds[idx] > genomeIndexPos {
```

Comment:
Is it more natural for this 'if' to come after the one below as an "else if" and lose the first clause?  In english it would be: if the insertion comes before the start, do this, else if it comes before the end, do this.